### PR TITLE
refactor: migrate test framework in build module

### PR DESCRIPTION
**Pull request by reidn3r**
These change are related to the issue #5555 

- Migrated from tap framework to node:test in fastify build module
- Ran the tests that i changed and confirmed they passed successfully

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,6 +133,11 @@
     {
       "name": "Aras Abbasi",
       "email": "aras.abbasi@gmail.com"
+    },
+    {
+      "name": "Reidner Adnan",
+      "url": "https://github.com/reidn3r",
+      "email": "rdn.adn00@gmail.com"
```
Comment (mcollina) [inline package.json:140]: Please revert this. That section of the package.json is left for the collaborators (also listed in the README).

**Review by mcollina — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (reidn3r) [reply package.json]: Of course! I misunderstood. I apologize.

**Comment by reidn3r** [comment]
@gurgunday  could you please review this PR?

**Review by gurgunday — COMMENTED** [review]
lgtm, but please uninstall tap

**Comment by jsumners** [comment]
> ~please uninstall tap~
> 
> Oh I see, I think we’d prefer a complete removal of tap in one PR rather than these incremental changes

I think that would be way too much. We _should_ go for incremental updates like this PR is proposing. But we cannot do it with the current way tests are run:

https://github.com/fastify/fastify/blob/7ddced69c45c294450ab12fe7f92f0d969fe86d7/package.json#L21-L30

I don't think there's a guarantee that using `tap` as a runner that it will recognize the output and result of `node:test` based suites. I think that before we can start on gradually migrating tests in this repo, we need to see a PR merged that looks a lot like https://github.com/newrelic/node-newrelic/pull/2387

**Comment by jsumners** [comment]
Why did you close this? #5720 should be merged soon.

**Pull request closed without merging**